admin/services/restart.service.js

```diff
@@ -200,12 +206,16 @@ restartService.markDeleted = async (pupilUrlSlug, userId, schoolId) => {
   const pupil = await pupilDataService.sqlFindOneBySlug(pupilUrlSlug, schoolId)
   const restart = await pupilRestartDataService.sqlFindOpenRestartForPupil(pupilUrlSlug, schoolId)
 
+  if (!restart) {
+    throw new Error('No restarts found to remove')
+  }
+
   // see if there is a check associated with this restart, ideally the slug
   // would refer to the restart itself, and not the pupil.
   if (restart.check_id) {
     const check = await pupilRestartDataService.sqlFindCheckById(restart.check_id, schoolId)
-    await checkStateService.changeState(check.checkCode, checkStateService.States.Expired)
-    await azureQueueService.addMessageAsync('prepared-check-delete', {
+    // await checkStateService.changeState(check.checkCode, checkStateService.States.Expired)
+    azureQueueService.addMessage('prepared-check-delete', {
```

Comment:
you will need to use the redis prepared check feature toggle here and drop from redis if true, otherwise send queue message